### PR TITLE
feat: create and delete passenger groups

### PR DESCRIPTION
**Pull request by RobCatton**
# Description

-   To allow the user to create groups from previously made passenger types, and also delete them groups.

# Testing instructions

-   Run locally and use the UI in global settings to create and delete passenger groups.

# Type of change

-   [x] feat - A new feature
-   [ ] fix - A bug fix
-   [ ] docs - Documentation only changes
-   [ ] codestyle - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] refactor - A code change that neither fixes a bug nor adds a feature
-   [ ] perf - A code change that improves performance
-   [ ] test - Adding missing tests or correcting existing tests
-   [ ] build - Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
-   [ ] ci - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] chore - Other changes that don't modify src or test files
-   [ ] revert - Reverts a previous commit
-   [ ] content - Changes to content or copy

### CODE REVIEW

File: repos/fdbt-site/src/pages/api/managePassengerTypes.ts [hunk 1]
```diff
@@ -93,8 +89,6 @@ export default async (req: NextApiRequestWithSession, res: NextApiResponse): Pro
 
         const nationalOperatorCode = getAndValidateNoc(req, res);
 
-        updateSessionAttribute(req, PASSENGER_TYPE_ATTRIBUTE, singlePassengerType.passengerType);
```
Comment (matt-twomey) [inline repos/fdbt-site/src/pages/api/managePassengerTypes.ts:96]: Why was this removed?

File: repos/fdbt-site/tests/pages/api/managePassengerGroup.test.ts [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+                ['Regular Senior-age-range-min']: '',
+                ['Regular Senior-age-range-max']: '',
+                ['Regular Senior-proof-docs']: [],
+                ['minimumPassengersRegular infant']: '1',
+                ['maximumPassengersRegular infant']: '2',
+                ['Regular infant-type']: 'infant',
+                ['Regular infant-age-range-min']: '1',
+                ['Regular infant-age-range-max']: '3',
+                ['Regular infant-proof-docs']: [],
+                passengerGroupName: 'My duplicate group',
```
Comment (matt-twomey) [inline repos/fdbt-site/tests/pages/api/managePassengerGroup.test.ts:226]: Not a big deal, but should we name this something else as it isn't a duplicate?

File: repos/fdbt-site/src/pages/api/managePassengerTypes.ts (hunk 1 above)
Comment (RobCatton) [reply repos/fdbt-site/src/pages/api/managePassengerTypes.ts:96]: It was unintended functionality - this is setting the users passenger type attribute in their session when they create one. It wasn't a requirement and needed removing.

File: repos/fdbt-site/tests/pages/api/managePassengerGroup.test.ts (hunk 2 above)
Comment (RobCatton) [reply repos/fdbt-site/tests/pages/api/managePassengerGroup.test.ts:226]: yeah sure! good catch

File: repos/fdbt-site/src/pages/api/managePassengerGroup.ts [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+                        errorMessage: 'Minimum amount cannot be greated than maximum amount',
+                        id: `minumum-passengers-${companion.name}`,
+                        userInput: companion.minNumber || '',
+                    });
+                }
+            }
+        });
+    }
+    const groupNameCheck = await getPassengerTypeByNameAndNocCode(nocCode, userInput.name, true);
+    const passengerTypeNameCheck = await getPassengerTypeByNameAndNocCode(nocCode, userInput.name, false);
```
[the remark this answers is not in the document]
Comment (RobCatton) [reply repos/fdbt-site/src/pages/api/managePassengerGroup.ts:114]: Yeah nice catch, could be an unnecessary DB call

File: repos/fdbt-site/src/pages/api/managePassengerGroup.ts [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+        passengerTypes.forEach((passengerType) => {
+            const minNumber = req.body[`minimumPassengers${passengerType}`];
+            const maxNumber = req.body[`maximumPassengers${passengerType}`];
+            const typeOfPassenger = req.body[`${passengerType}-type`];
+            const ageRangeMin = req.body[`${passengerType}-age-range-min`];
+            const ageRangeMax = req.body[`${passengerType}-age-range-max`];
+            const proofDocuments = req.body[`${passengerType}-proof-docs`];
+            companions.push({
+                name: passengerType,
+                passengerType: typeOfPassenger,
```
[the remark this answers is not in the document]
Comment (RobCatton) [reply repos/fdbt-site/src/pages/api/managePassengerGroup.ts:32]: yep nice catch

File: repos/fdbt-site/src/interfaces/index.ts [hunk 5]
```diff
@@ -201,6 +201,11 @@ export interface GlobalSettingsCounts {
     // More to be added
 }
 
+export interface GlobalSettingsGroupInputs {
```
[the remark this answers is not in the document]
Comment (RobCatton) [reply repos/fdbt-site/src/interfaces/index.ts:204]: Forgot about that format, I'll change now.

File: repos/fdbt-site/src/pages/api/managePassengerGroup.ts [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+    const groupName = removeExcessWhiteSpace(req.body.passengerGroupName);
+    const passengerTypes: string[] = !req.body.passengerTypes
+        ? []
+        : isArray(req.body.passengerTypes)
+        ? req.body.passengerTypes
+        : [req.body.passengerTypes];
+
+    const companions: CompanionInfo[] = [];
+    if (passengerTypes.length > 0) {
+        passengerTypes.forEach((passengerType) => {
```
Comment (jacsamell) [inline repos/fdbt-site/src/pages/api/managePassengerGroup.ts:23]: Minor: this could be done with a map to create the companions array from the passengerTypes array to save on the pushing and external initialisation of the empty array

File: repos/fdbt-site/src/pages/api/managePassengerGroup.ts [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+    if (groupNameCheck) {
+        errors.push({
+            errorMessage: 'There is already a group with this name. Choose another',
+            id: 'passenger-group-name',
+            userInput: userInput.name,
+        });
+        return errors;
+    }
+
+    if (passengerTypeNameCheck) {
```
Comment (jacsamell) [inline repos/fdbt-site/src/pages/api/managePassengerGroup.ts:125]: As discussed I'm not sure if we should be checking this?

File: repos/fdbt-site/src/pages/api/managePassengerGroup.ts [hunk 8]
```diff
@@ -12,32 +12,30 @@ import { checkIntegerIsValid, removeExcessWhiteSpace } from './apiUtils/validato
...
-                ageRangeMin,
-                ageRangeMax,
-                proofDocuments,
-            });
-        });
-    }
+    const companions: CompanionInfo[] = passengerTypeNames.map((passengerTypeName) => {
+        const minNumber = req.body[`minimumPassengers${passengerTypeName}`];
+        const maxNumber = req.body[`maximumPassengers${passengerTypeName}`];
+        const typeOfPassenger = req.body[`${passengerTypeName}-type`];
```
Comment (matt-twomey) [inline repos/fdbt-site/src/pages/api/managePassengerGroup.ts:24]: Could this be called 'passengerType' just to avoid needing to rename it on line 31?

**Review by matt-twomey — APPROVED** [review]
